libs/webb-ui-components/src/components/SideBar/components/SubItem.tsx

```diff
@@ -0,0 +1,73 @@
+import { ExternalLinkLine } from '@webb-tools/icons';
+import React, { useMemo } from 'react';
+import { pushToInternalLink, pushToExternalLink } from '../utils';
+import { Typography } from '../../../typography/Typography';
+import { twMerge } from 'tailwind-merge';
+
+export type SubItemProps = {
+  name: string;
+  isInternal: boolean;
+  href: string;
+};
+
+export type ExtraSubItemProps = {
+  isActive?: boolean;
+  setItemIsActive?: () => void;
+  setSubItemIsActive?: () => void;
+};
+
+export const SubItem: React.FC<SubItemProps & ExtraSubItemProps> = ({
+  name,
+  isInternal,
+  href,
+  isActive,
+  setItemIsActive,
+  setSubItemIsActive,
+}) => {
+  const pushToLinkAndSetIsActive = () => {
+    if (setItemIsActive && setSubItemIsActive && isInternal) {
+      setItemIsActive();
+      setSubItemIsActive();
+    }
+
+    if (isInternal && href) {
+      pushToInternalLink(href);
+    } else if (!isInternal && href) {
+      pushToExternalLink(href);
+    }
+  };
```

Comment:
I think this is not really good for accessibility, we should use the anchor tag `<a></a>` for links instead of `window.open` here.